### DISCUSSION TITLE
suggestion: adhere to solidity style guide

### DISCUSSION THREAD

**Discussion by n4n0b1t3** [smartcontractkit/full-blockchain-solidity-course-py#873] [Ideas]
Maybe it's just me, but I find it helpful to always follow the same order when adding modifiers. This is nothing of importance, maybe a small improvement. Perhaps, following the solidity style guide would be a start.
The modifier order for a function should be:
- Visibility
- Mutability
- Virtual
- Override
- Custom modifiers

https://docs.soliditylang.org/en/v0.8.11/style-guide.html?highlight=style%20guide#function-declaration

**Comment by cromewar** [comment]
Nice hint @n4n0b1t3 this is great, do you thing is worth it to add it somehow to the repo index @PatrickAlphaC ?

**Reply by n4n0b1t3** [reply]
For consistency's sake, I would suggest first going through the boot camp examples, which are most frequently used by beginners, and make sure those adhere to the guide.  With "repo index" do you refer to the README.md? Perhaps a "beginner best practice for solidity programming novices" section could be a useful extension.

**Reply by cromewar** [reply]
You have a great idea here, let's see what Patrick things about this.

**Reply by PatrickAlphaC** [reply]
Agreed! Yeah I meant to add the style guide in there too... but I sort of forgot to 😅

Thanks for this! I'll add it to the hardhat one!
